primitiv/c/internal.h

```diff
@@ -36,6 +36,9 @@ inline const primitiv::cpp_name *to_cpp_ptr(const c_name *instance) { \
 #define PRIMITIV_C_VAL_TO_PTR(cpp_name, c_name) \
 inline c_name *to_c_ptr_from_value(const primitiv::cpp_name &instance) { \
   return reinterpret_cast<c_name*>(new primitiv::cpp_name(instance)); \
+} \
```

Comment:
Probably these 2 functions can be integrated using [*perfect forwarding*](http://en.cppreference.com/w/cpp/utility/forward).